pkg/apis/autoscaling/v1alpha1/federatedhpa_types.go

```diff
@@ -0,0 +1,74 @@
+package v1alpha1
+
+import (
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:shortName=fhpa,categories={karmada-io}
+
+// FederatedHPA is centralized HPA that can aggregate the metrics in multiple clusters.
+// When the system load increases, it will query the metrics from multiple clusters and scales up the replicas.
+// When the system load decreases, it will query the metrics from multiple clusters and scales down the replicas.
+// After the replicas are scaled up/down, karmada-scheduler will schedule the replicas based on the policy.
+type FederatedHPA struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the specification of the HPAAggregator.
```

Comment:
Some comments have not changed.
```suggestion
	// Spec is the specification of the FederatedHPA.
```